dali/test/python/test_operator_erase.py

```diff
@@ -0,0 +1,239 @@
+            shape_0 = shape[n*region_length+0]
+            end_0 = start_0 + shape_0
+
+            if region_length > 1:
+                start_1 = anchor[n*2+1]
+                shape_1 = shape[n*2+1]
+                end_1 = start_1 + shape_1
+
+            if axis_names == "H":
+                assert H >= end_0
+                image[start_0:end_0, :, :] = 0
+            elif axis_names == "W":
+                assert W >= end_0
+                image[:, start_0:end_0, :] = 0
+            elif axis_names == "HW":
+                assert H >= end_0
+                assert W >= end_1
+                image[start_0:end_0, start_1:end_1, :] = 0
+            elif axis_names == "WH":
+                assert H >= end_1
+                assert W >= end_0
+                image[start_1:end_1, start_0:end_0, :] = 0
+            else:
+                assert(False)  # should not happen
+        return image
+    else:
+        assert(False)  # should not happen
+
+class ErasePythonPipeline(Pipeline):
+    def __init__(self, function, batch_size, data_layout, iterator,
+                 anchor, shape, axis_names, axes,
+                 erase_func=erase_func,
+                 num_threads=1, device_id=0):
+        super(ErasePythonPipeline, self).__init__(batch_size,
+                                                  num_threads,
+                                                  device_id,
+                                                  exec_async=False,
+                                                  exec_pipelined=False)
+        self.iterator = iterator
+        self.inputs = ops.ExternalSource()
+        self.data_layout = data_layout
+
+        function = partial(erase_func, anchor, shape, axis_names, axes, data_layout)
+
+        self.erase = ops.PythonFunction(function=function)
+
+    def define_graph(self):
+        self.data = self.inputs()
+        out = self.erase(self.data)
+        return out
+
+    def iter_setup(self):
+        data = self.iterator.next()
+        self.feed_input(self.data, data, layout=self.data_layout)
+
+
+def check_operator_erase_vs_python(device, batch_size, input_shape,
+                                   anchor, shape, axis_names, axes, input_layout):
+    eii1 = RandomDataIterator(batch_size, shape=input_shape, dtype=np.float32)
+    eii2 = RandomDataIterator(batch_size, shape=input_shape, dtype=np.float32)
+    compare_pipelines(
+        ErasePipeline(device, batch_size, input_layout, iter(eii1),
+                      anchor=anchor, shape=shape, axis_names=axis_names, axes=axes),
+        ErasePythonPipeline(device, batch_size, input_layout, iter(eii2),
+                            anchor=anchor, shape=shape, axis_names=axis_names, axes=axes),
+        batch_size=batch_size, N_iterations=5, eps=1e-04)
+
+
+def test_operator_erase_vs_python():
+    # layout, shape, axis_names, axes, anchor, shape
+    rois = [("HWC", (60, 80, 3), "HW", None, (4, 10), (40, 50)),
```

Comment:
Can you add a case for volumetric data and 1c images?

Reply:
done